libwayshot/src/lib.rs

```diff
@@ -446,13 +447,50 @@ impl WayshotConnection {
             }
         };
 
+        let shm = self.globals.bind::<WlShm, _, _>(&qh, 1..=1, ())?;
         for (frame_copy, frame_guard, output_info) in frames {
             tracing::span!(
                 tracing::Level::DEBUG,
                 "overlay_frames::surface",
                 output = format!("{output_info}")
             )
             .in_scope(|| -> Result<()> {
+                let buffer = {
+                    let output_size = output_info.logical_region.inner.size;
+                    if output_size != frame_copy.frame_format.size {
+                        let file = tempfile::tempfile().unwrap();
```

Comment:
```suggestion
                        let file = tempfile::tempfile()?;
```

If not the correct error type, wrap it in another type, don't unwrap.